pkg/action/lint.go

```diff
@@ -94,12 +94,15 @@ func lintChart(path string, vals map[string]interface{}, namespace string, stric
 			return linter, errors.Wrap(err, "unable to extract tarball")
 		}
 
-		lastHyphenIndex := strings.LastIndex(filepath.Base(path), "-")
-		if lastHyphenIndex <= 0 {
-			return linter, errors.Errorf("unable to parse chart archive %q, missing '-'", filepath.Base(path))
+		files, err := ioutil.ReadDir(tempDir)
+		if err != nil {
+			return linter, errors.Wrapf(err, "unable to read temporary output directory %s", tempDir)
+		}
+		if !files[0].IsDir() {
```

Comment:
Should length  be checked here?

Reply:
It was not originally, but it probably should. Fixing now

Reply:
So actually, it turns out that the tarball logic errors if no files exist in the chart archive, so no need to check it here